src/realm/util/file.cpp

```diff
@@ -198,6 +208,18 @@ void remove_dir_recursive(const std::string& path)
 }
 
 
+bool try_remove_dir_recursive(const std::string& path)
+{
+    try {
+        remove_dir_recursive(path);
+    }
+    catch (const File::NotFound&) {
+        return false;
+    }
+    return true;
+}
```

Comment:
I will change to your suggestion. 
I avoided that because I found it double effect to first check for existence and then
remove the directory.
But that is fine with me.